cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -362,15 +369,45 @@ neighbor_sample_impl(raft::handle_t const& handle,
     level_result_label_vectors = std::nullopt;
   }
 
-  return detail::shuffle_and_organize_output(handle,
-                                             std::move(result_srcs),
-                                             std::move(result_dsts),
-                                             std::move(result_weights),
-                                             std::move(result_edge_ids),
-                                             std::move(result_edge_types),
-                                             std::move(result_hops),
-                                             std::move(result_labels),
-                                             label_to_output_comm_rank);
+  std::optional<rmm::device_uvector<size_t>> result_offsets{std::nullopt};
+
+  std::tie(result_srcs,
+           result_dsts,
+           result_weights,
+           result_edge_ids,
+           result_edge_types,
+           result_hops,
+           result_labels,
+           result_offsets) = detail::shuffle_and_organize_output(handle,
+                                                                 std::move(result_srcs),
+                                                                 std::move(result_dsts),
+                                                                 std::move(result_weights),
+                                                                 std::move(result_edge_ids),
+                                                                 std::move(result_edge_types),
+                                                                 std::move(result_hops),
+                                                                 std::move(result_labels),
+                                                                 label_to_output_comm_rank);
+
+  if (result_srcs.size() == 0) {
+    // Update the 'edgelist_label_offsets' array to be proportional to the
+    // number of labels
+    result_offsets->resize(num_labels + 1, handle.get_stream());
```

Comment:
Is there any guarantee that if results_srcs.size() != 0, result_offsets.size() == num_labels + 1?  Say, we have 100 labels but only one label has a sampled edge, will this still work?

Reply:
> Say, we have 100 labels but only one label has a sampled edge, will this still work?

This is a valid edge case to consider. I believe the current implementation of neighborhood sampling only keeps track of vertices that have outgoing edges. @ChuckHastings any thought?